server/src/test/java/com/octopus/teamcity/opentelemetry/server/OTELHelperTest.java

```diff
@@ -46,39 +47,93 @@ void isReadyTest() {
+
+        // Assert
+        assertEquals(expectedSpan, actualSpan);
     }
 
     @Test
-    void createTransientSpanTest(@Mock SRunningBuild build) {
-        Span parentSpan = this.otelHelper.getParentSpan(String.valueOf(build.getBuildId()));
-        assertNotNull(this.otelHelper.createTransientSpan(String.valueOf(build.getBuildId()), parentSpan, new Date().getTime()));
+    void createTransientSpanShouldReturnSpanAndNotBeAvailableInGetSpan() {
+        // Arrange
+        SRunningBuild build = mock(SRunningBuild.class, RETURNS_DEEP_STUBS);
+        SRunningBuild parentBuild = mock(SRunningBuild.class, RETURNS_DEEP_STUBS);
+        Span parentSpan = createParentSpanForTest(parentBuild);
+        // Work around for mocked build.getBuildId(), as you cannot mock a final method (will return 0 in all cases when called), this ensures that the parent build id and the build id are different values
+        // This ensures two different spans are created, a build span and a parent span
+        when(build.getBuildId()).thenReturn(12345L);
+        String buildId = String.valueOf(build.getBuildId());
+
+        // Act
+        Span expectedSpan = this.otelHelper.createTransientSpan(buildId, parentSpan, new Date().getTime());
+
+        // Assert
+        assertNotNull(expectedSpan);
+        assertNotNull(String.valueOf(parentBuild.getBuildId()));
+        assertNull(this.otelHelper.getSpan(buildId));
     }
 
     @Test
-    void removeSpanTest(@Mock SRunningBuild build) {
-        Span parentSpan = this.otelHelper.getParentSpan(String.valueOf(build.getBuildId()));
+    void aRemovedSpanShouldNotBeAvailableInGetSpan() {
+        // Arrange
+        SRunningBuild build = mock(SRunningBuild.class, RETURNS_DEEP_STUBS);
+        Span parentSpan = createParentSpanForTest(build);
         this.otelHelper.createSpan(String.valueOf(build.getBuildId()), parentSpan);
+
+        // Act
         this.otelHelper.removeSpan(String.valueOf(build.getBuildId()));
+
+        // Assert
         assertNull(this.otelHelper.getSpan(String.valueOf(build.getBuildId())));
     }
 
     @Test
-    void getSpanExistsTest(@Mock SRunningBuild build) {
-        Span parentSpan = this.otelHelper.getParentSpan(String.valueOf(build.getBuildId()));
+    void getSpanShouldReturnSpanIfExists() {
+        // Arrange
+        SRunningBuild build = mock(SRunningBuild.class, RETURNS_DEEP_STUBS);
+        Span parentSpan = createParentSpanForTest(build);
         this.otelHelper.createSpan(String.valueOf(build.getBuildId()), parentSpan);
-        assertNotNull(this.otelHelper.getSpan(String.valueOf(build.getBuildId())));
+
+        // Act
+        Span expectedSpan = this.otelHelper.getSpan(String.valueOf(build.getBuildId()));
+
+        // Assert
+        assertNotNull(expectedSpan);
     }
 
     @Test
-    void getSpanNotExistsTest(@Mock SRunningBuild build) {
+    void getSpanShouldNOTReturnSpanWhenItDoesNotExist(@Mock SRunningBuild build) {
+        // Assert
```

Comment:
```suggestion
        // Act & Assert
```
Consider commenting that the `otelHelper` is new for this test so is expected to be empty

Reply:
done!